src/main/java/org/jitsi/jicofo/bridge/BridgeSelectionStrategy.java

```diff
@@ -86,61 +93,216 @@ public Bridge select(
     }
 
     /**
-     * Selects a bridge to be used for a new participant in a conference,
-     * assuming that no other bridge is used by the conference (i.e. this
-     * is the initial selection of a bridge for the conference).
+     * A1. (Happy case 1): There is a NOL bridge in the conference and in the
+     * region. Use the least loaded of them.
      *
-     * @param bridges the list of bridges to select from.
-     * @param participantRegion the region of the participant for which a
-     * bridge is to be selected.
-     * @return the selected bridge, or {@code null} if no bridge is
-     * available.
+     * @param bridges
+     * @param conferenceBridges
+     * @param participantRegion
+     *
+     * @return
      */
-    private Bridge selectInitial(List<Bridge> bridges,
-                                 String participantRegion)
+    private Optional<Bridge> findNotLoadedBridgeAlreadyInConferenceInRegion(List<Bridge> bridges,
+                                                                            List<Bridge> conferenceBridges,
+                                                                            String participantRegion)
     {
-        Bridge bridge = null;
+        Optional<Bridge> result = bridges.stream()
+            .filter(not(Bridge::isOverloaded))
+            .filter(selectFrom(conferenceBridges))
+            .filter(inRegion(participantRegion))
+            .findFirst();
 
-        // Prefer a bridge in the participant's region.
-        if (participantRegion != null)
+        if (result.isPresent())
         {
-            bridge = findFirstOperationalInRegion(bridges, participantRegion);
+            logger.info("bridge_selected,rule=a1,bridge=" + result.get() +
+                ",conference_bridges=" + conferenceBridges.stream().map(Bridge::toString).collect(Collectors.joining(", ")) +
+                "participant_region " + participantRegion);
+
+            totalA1++;
         }
 
-        // Otherwise, prefer a bridge in the local region.
-        if (bridge == null)
+        return result;
+    }
+
+    /**
+     * A2. (Happy case 2): There is a NOL bridge in the region, and the
+     * conference has no bridges in the region. Use the least loaded bridge in
+     * the region.
+     *
+     * @param bridges
+     * @param participantRegion
+     *
+     * @return
+     */
+    private Optional<Bridge> findNotLoadedBridgeInRegion(List<Bridge> bridges,
+                                                         List<Bridge> conferenceBridges,
+                                                         String participantRegion)
+    {
+        // NOTE that A2 is equivalent to B1 but we include it as a separate
```

Comment:
Is this comment still relevant?  Looks like all references to B1 can go away (and I think we should move away from the A1, A2, etc. references in general)